common/src/uint/impls.rs

```diff
@@ -542,8 +542,8 @@ macro_rules! construct_uint {
             fn fmt(&self, f: &mut ::core::fmt::Formatter) -> ::core::fmt::Result {
                 let &$name(ref data) = self;
                 write!(f, "0x")?;
-                for ch in data.iter().rev() {
-                    write!(f, "{:016x}", ch)?;
+                for ch in data {
+                    write!(f, "{:016x}", ch.to_be())?;
```

Comment:
I don't think [to_be](https://doc.rust-lang.org/std/primitive.u64.html#method.to_be) does what you want here. It is a platform-dependent operation that does nothing on a BE computer and swaps bytes on a LE computer (causing it to print essentially a completely different number). So different string will be printed depending on endianness of the machine the code is running on, which does not sound right.

Reply:
You're saying that it's machine dependent, as if it's a problem. I really don't understand. Don't we want a specific output format regardless of the machine type? Whether it's big or little endian, it'll be formatted as big endian. What's wrong with that?

Reply:
I think `u64::to_be` will not work correctly on big endian (which is no-op there) and `u64::swap_bytes` should be used instead.

Reply:
> You're saying that it's machine dependent, as if it's a problem. I really don't understand. Don't we want a specific output format regardless of the machine type? Whether it's big or little endian, it'll be formatted as big endian. What's wrong with that?

The default formatting routine already takes care of that. `write!(f, "{:016x}", ch)` already writes the number as a 64-bit number regardless of the platform. So if you explicitly convert `to_be`, the conversion happens in effect twice, so you end up printing in native order which is platform dependent.

Reply:
I reverted this change as per our discussion.